server/util/cosmos.js

```diff
@@ -26,34 +26,41 @@ import {
 } from '../config/config';
 import { COSMOS_PRIVATE_KEY } from '../config/secret';
 
-let wallet;
-let delegatorAddress;
-let queryClient;
-let signingClient;
-
-(() => async function createSigner() {
-  wallet = await DirectSecp256k1Wallet.fromKey(COSMOS_PRIVATE_KEY);
-  const [firstAccount] = await wallet.getAccounts();
-  delegatorAddress = firstAccount.address;
-  const tendermint34Client = await Tendermint34Client.connect(cosmosRPCEndpoint);
-  queryClient = QueryClient.withExtensions(
-    tendermint34Client,
-    setupAuthExtension,
-    setupBankExtension,
-  );
-  signingClient = await SigningStargateClient.connectWithSigner(cosmosRPCEndpoint, wallet);
+const getCosmos = (() => {
+  let cosmos;
+  return async () => {
+    if (!cosmos) {
+      const wallet = await DirectSecp256k1Wallet.fromKey(COSMOS_PRIVATE_KEY);
```

Comment:
Concept source of this sophisticated async lazy initializer [here](https://stackoverflow.com/questions/58796320/what-is-a-canonical-safe-way-of-async-lazy-initialization-in-javascript/58796336#58796336).